src/android/com/fivestars/mtab/plugin/RestartAppPlugin.java

```diff
@@ -64,21 +69,34 @@ private boolean restartApp(long delay, boolean exit) {
             Log.d("RestartAppPlugin", "Launch Intent not found");
             return false;
         }
-        
-        intent.addFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP);
 
-        cordovaActivity.finish();
+        intent.addFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP);
 
+        mPendingRestart = PendingIntent.getActivity(appContext, 0, intent, PendingIntent.FLAG_CANCEL_CURRENT);
         // Pending intent starts activity after 2 seconds
         AlarmManager mgr = (AlarmManager)appContext.getSystemService(Context.ALARM_SERVICE);
-        mgr.set(AlarmManager.RTC, System.currentTimeMillis() + delay, PendingIntent.getActivity(appContext, 0, intent, 0));
+        mgr.set(AlarmManager.RTC, System.currentTimeMillis() + delay, mPendingRestart);
 
         if (exit) {
             // ToDo (bbil): Remove this once once cordova webviews are properly destroyed
             // Either cordova must be upgraded or webviews manually destroyed
+            cordovaActivity.finish();
+
             System.exit(2);
         }
 
         return true;
     }
+
+    private boolean cancelPendingRestart() {
+        if (mPendingRestart != null) {
+            try {
+                mPendingRestart.cancel();
+                return true;
+            } catch (Exception e) {
+                Log.w("RestartAppPlugin", "Could not cancel pending restart");

```

Comment:
I think it might be better to log the exception so you get the stack trace and to do it at the error level.
`Log.e("RestartAppPlugin", "Could not cancel pending restart", e)`